include/swift/AST/ConstTypeInfo.h

```diff
@@ -55,13 +56,9 @@ class RawLiteralValue : public CompileTimeValue {
 };
 
 struct FunctionParameter {
-public:
-  std::string getLabel() { return Label; }
-  swift::Type *getType() { return Type; }
-
-private:
   std::string Label;
-  swift::Type *Type;
+  swift::Type Type;
+  std::shared_ptr<CompileTimeValue> Value;
```

Comment:
What's the need for this to be `shared` instead of `unique`?

Reply:
`ConstValueTypePropertyInfo` used `shared` which allowed me to use the same type for the `writeValue(llvm::json::OStream &JSON, std::shared_ptr<CompileTimeValue> Value)` function.

I may be able to use a `unique_ptr` here and covert `writeValue` to take a direct `CompileTimeValue` instead?

Is the downside of using `shared` the performance impact?

Reply:
Ah, I see. Makes sense to keep it as-is then. 
The downside I was thinking of is the general increase in difficulty of tracking ownership, but since we have precedent already, then this makes sense, thanks.